src/main/java/io/vertx/core/impl/launcher/CommandLineUtils.java

```diff
@@ -0,0 +1,77 @@
+/*
+ *  Copyright (c) 2011-2015 The original author or authors
+ *  ------------------------------------------------------
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *       The Eclipse Public License is available at
+ *       http://www.eclipse.org/legal/epl-v10.html
+ *
+ *       The Apache License v2.0 is available at
+ *       http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+
+package io.vertx.core.impl.launcher;
+
+import java.io.File;
+
+/**
+ * Utilities method to analyse original command line.
+ *
+ * @author Clement Escoffier <clement@apache.org>
+ */
+public class CommandLineUtils {
+
+  /**
+   * @return the fat-jar file used to execute the application if the fat-jar approach is used.
+   */
+  public static String getJar() {
+    // Check whether or not the "sun.java.command" system property is defined,
+    // if it is, check whether the first segment of the command ends with ".jar".
+    final String command = getCommand();
```

Comment:
I think you can use `getFirstSegmentOfCommand ()` in the same class to achieve the same thing.